packages/mapviewer/src/utils/composables/useMovableElement.composable.js

```diff
@@ -171,36 +134,25 @@ export function useMovableElement(element, options = {}) {
      * the drag and drop useless (stuck). We make sure the initial position of the element is valid
      * here.
      */
-    function constrainsElementWithinViewport() {
-        const currentPosition = {
-            left: toValue(element).offsetLeft,
-            top: toValue(element).offsetTop,
-        }
-        const elementSize = toValue(element).getBoundingClientRect()
-        const positionConstrainedByNewLimits = {
-            ...currentPosition,
-        }
-        if (viewport.value.top > currentPosition.top) {
-            positionConstrainedByNewLimits.top = viewport.value.top
-        }
-        if (viewport.value.bottom < currentPosition.top + elementSize.height) {
-            positionConstrainedByNewLimits.top = viewport.value.bottom - elementSize.height
-        }
-        if (viewport.value.left > currentPosition.left) {
-            positionConstrainedByNewLimits.left = viewport.value.left
-        }
-        if (viewport.value.right < currentPosition.left + elementSize.width) {
-            positionConstrainedByNewLimits.right = viewport.value.right - elementSize.width
-        }
-        if (
-            currentPosition.top !== positionConstrainedByNewLimits.top ||
-            currentPosition.left !== positionConstrainedByNewLimits.left
-        ) {
-            placeElementAt(
-                positionConstrainedByNewLimits.top,
-                positionConstrainedByNewLimits.left,
-                MovementSource.WINDOW_RESIZE
-            )
-        }
+    function constrainElementWithinViewport(deltaX = 0, deltaY = 0) {
```

Comment:
I would not put default values here, and add the `0, 0` as parameter in the initialize call. This forces us to be explicit in tests and in the code.

I would also rework the comment a  bit, as we now also handle placing an element and not only ensuring it's within bounds at startup.